core/src/env/xrml/calls.rs

```diff
@@ -0,0 +1,236 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::env::EnvTypes;
+use core::convert::TryInto;
+use scale::{
+    Decode,
+    Encode,
+    Output,
+};
+
+#[cfg(feature = "old-codec")]
+#[cfg_attr(feature = "test-env", derive(Debug, Clone, PartialEq, Eq))]
+pub enum Address<T: EnvTypes, AccountIndex> {
+    Id(T::AccountId),
+    Index(AccountIndex),
+}
+
+#[cfg(feature = "old-codec")]
+fn need_more_than<T: PartialOrd>(a: T, b: T) -> Option<T> {
+    if a < b {
+        Some(b)
+    } else {
+        None
+    }
+}
+
+/// Decode implementation copied over from Substrate `Address` that can be found [here](substrate-address).
+///
+/// # Note
+/// This implementation MUST be kept in sync with substrate, tests below will ensure that.
+///
+/// [substrate-address]: https://github.com/paritytech/substrate/blob/ec62d24c602912f07bbc416711376d9b8e5782c5/srml/indices/src/address.rs#L61
+#[cfg(feature = "old-codec")]
+impl<T, AccountIndex> scale::Decode for Address<T, AccountIndex>
+where
+    T: EnvTypes,
+    AccountIndex: scale::Decode + From<u32> + PartialOrd + Copy + Clone,
+{
+    fn decode<I: scale::Input>(input: &mut I) -> Option<Self> {
+        Some(match input.read_byte()? {
+            x @ 0x00..=0xef => Address::Index(AccountIndex::from(x as u32)),
+            0xfc => {
+                Address::Index(AccountIndex::from(need_more_than(
+                    0xef,
+                    u16::decode(input)? as u32,
+                )?))
+            }
+            0xfd => {
+                Address::Index(AccountIndex::from(need_more_than(
+                    0xffff,
+                    u32::decode(input)?,
+                )?))
+            }
+            0xfe => {
+                Address::Index(need_more_than(
+                    AccountIndex::from(0xffffffffu32),
+                    Decode::decode(input)?,
+                )?)
+            }
+            0xff => Address::Id(Decode::decode(input)?),
+            _ => return None,
+        })
+    }
+}
+
+/// Encode implementation copied over from Substrate `Address` that can be found [here](substrate-address).
+///
+/// # Note
+/// This implementation MUST be kept in sync with substrate, tests below will ensure that.
+///
+/// [substrate-address]: https://github.com/paritytech/substrate/blob/ec62d24c602912f07bbc416711376d9b8e5782c5/srml/indices/src/address.rs#L83
+impl<T, AccountIndex> Encode for Address<T, AccountIndex>
```

Comment:
```suggestion
impl<T, AccountIndex> scale::Encode for Address<T, AccountIndex>
```
or 

change `Decode` format to 

```
impl<T, AccountIndex> Decode for Address<T, AccountIndex>
```
Keep the `Encode/Decode` format consistent.